pkg/kritis/secrets/secrets.go

```diff
@@ -48,17 +59,31 @@ func Fetch(namespace string, name string) (*PGPSigningSecret, error) {
 	if err != nil {
 		return nil, err
 	}
-	pub, ok := secret.Data[constants.PublicKey]
+	pub, ok := secret.Data[PublicKey]
 	if !ok {
-		return nil, fmt.Errorf("invalid secret %s. could not find key %s", name, constants.PublicKey)
+		return nil, fmt.Errorf("invalid secret %s. could not find key %s", name, PublicKey)
 	}
-	priv, ok := secret.Data[constants.PrivateKey]
+	priv, ok := secret.Data[PrivateKey]
 	if !ok {
-		return nil, fmt.Errorf("invalid secret %s. could not find key %s", name, constants.PublicKey)
+		return nil, fmt.Errorf("invalid secret %s. could not find key %s", name, PrivateKey)
+	}
+	pb, ok := secret.Data[Passphrase]
+	phrase := ""
+	if ok {
+		// Passphrase was provided
+		// Verify the passphrase is base64 encoded
+		dec, err := base64.StdEncoding.DecodeString(string(pb))
+		if err != nil {
+			return nil, fmt.Errorf("passphrase was not base64 encoded")
```

Comment:
Preserve the original err. It seems dubious to assert from the existence of an error that the passphrase was not base64 encoded: it could have failed for other reasons. However, Do continue to add context to the error message so that regular users and developers can immediately understand its source, for instance:

return fmt.Errorf("base64 decode failed: %v", err)

When the error is ultimately handled by the program's main function, it should provide a clear causal chain from the root problem to the overall failures reminiscent of a NASA accident investigation:

genesis: crashed: no parachute: G-switch failed: bad relay orientation

Reply:
:) that's a great point, fixed. thanks!